var/spack/repos/builtin/packages/swig/package.py

```diff
@@ -70,3 +70,41 @@ def create_symlink(self):
         @when(_version)
         def autoreconf(self, spec, prefix):
             which('sh')('./autogen.sh')
+
+
+    @property
+    def _installed_exe(self):
+        return join_path(self.prefix, 'bin', 'swig')
+
+    def _test_version(self):
+        version = str(self.version)
+        if version.endswith('-fortran'):
+            version = version.replace('-', r'\+')
+        elif version in ('fortran', 'master'):
+            version = r'.*'
+
+        self.run_test(self._installed_exe,
+                      '-version',
+                      expected=[r'SWIG Version {0}'.format(version)],
+                      purpose="test: version")
+
+    def _test_swiglib(self):
+        swig = Executable(self._installed_exe)
+        swiglib = swig('-swiglib', output=str).strip()
```

Comment:
Why not use `self.run_test`?

Reply:
`run_test` just checks the output against a regex (and checks error codes). Here I'm using the actual output from the executable call to test for the existence of installed components.